enhancements/network/on-prem-dual-stack-vips.md

```diff
@@ -258,6 +265,21 @@ will be maintained. We will not (and cannot) automatically add new VIPs to a
 cluster on upgrade. If a deployer of an older dual stack cluster wants the new
 VIP functionality that will have to be a separate operation from upgrade.
 
+CNO takes care updating the new fields (`apiServerInternalIPs` &
+`ingressInternalIPs`) and old fields (`apiServerInternalIP` &
+`ingressInternalIP`) in [openshift/api](#openshiftapi) to have a consistent API
+between versions and therefore keeping clients consuming the old API functional.
+
+The following table shows the rules how the fields are set:
+
+| Case | Initial value of new field | Initial value of old field | Resulting value of new field | Resulting value of old field | Description |
+| ---- | -------------------------- | -------------------------- | ---------------------------- | ---------------------------- | ----------- |
+| 1    | _empty_                    | foo                        | [0]: foo                     | foo                          | `new` field is empty, `old` with value: set `new[0]` to value from `old` |
+| 2    | [0]: foo <br />[1]: bar    | _empty_                    | [0]: foo <br />[1]: bar      | foo                          | `new` contains values, `old` is empty: set `old` to value from `new[0]` |
+| 3    | [0]: foo <br />[1]: bar    | foo                        | [0]: foo <br />[1]: bar      | foo                          | `new` field contains values, `old` contains `new[0]`: we are fine, as `old` is part of `new` |
+| 4    | [0]: foo <br />[1]: bar    | bar                        | [0]: foo <br />[1]: bar      | bar                          | `new` contains values, `old` contains `new[1]`: we are fine, as `old` is part of `new` |
```

Comment:
The original `apiServerInternalIP` and `ingressInternalIP` values had to have been the cluster's primary IP family, so `old` should not be `new[1]`... either `old` is wrong or `new` is in the wrong order.